benchmarks/serialize.py

```diff
@@ -0,0 +1,246 @@
+            hash_engine(),  # pytype: disable=not-instantiable
+            chunk_size=chunk_size,
+        )
+
+    return _hasher_factory
+
+
+def run(args: argparse.Namespace) -> None:
+    """Performs the benchmark.
+
+    Args:
+        args: The arguments specifying the benchmark scenario.
+    """
+    # 1. Hashing layer
+    if args.use_shards:
+        hasher = get_sharded_file_hasher_factory(
+            args.hash_method, args.chunk, args.shard
+        )
+    else:
+        hasher = get_file_hasher_factory(args.hash_method, args.chunk)
+
+    # 2. Serialization layer
+    if args.skip_manifest or args.single_digest:
+        merge_hasher_factory = get_hash_engine_factory(args.merge_hasher)
+        if args.use_shards:
+            serializer = serialize_by_file_shard.DigestSerializer(
+                hasher,
+                merge_hasher_factory(),  # pytype: disable=not-instantiable
+                max_workers=args.max_workers,
+            )
+        else:
+            # This gets complicated because the API here is not matching the
+            # rest. We should fix this.
+            if args.max_workers is not None and args.max_workers != 1:
+                raise ValueError("Currently, only 1 worker is supported here")
+            serializer = serialize_by_file.DigestSerializer(
+                # pytype: disable=wrong-arg-count
+                hasher(pathlib.Path("unused")),
+                # pytype: enable=wrong-arg-count
+                merge_hasher_factory,
+            )
+    else:
+        if args.use_shards:
+            serializer_factory = serialize_by_file_shard.ManifestSerializer
+        else:
+            serializer_factory = serialize_by_file.ManifestSerializer
+
+        serializer = serializer_factory(hasher, max_workers=args.max_workers)
+
+    # 3. Signing layer
+    if args.skip_manifest:
+        in_toto_builder = id
+    else:
+        if args.single_digest:
+            in_toto_builder = in_toto.SingleDigestIntotoPayload
+        else:
+            if args.digest_of_digests:
+                if args.use_shards:
+                    in_toto_builder = in_toto.DigestOfShardDigestsIntotoPayload
+                else:
+                    in_toto_builder = in_toto.DigestOfDigestsIntotoPayload
+            else:
+                if args.use_shards:
+                    in_toto_builder = in_toto.ShardDigestsIntotoPayload
+                else:
+                    in_toto_builder = in_toto.DigestsIntotoPayload
+
+        in_toto_builder = in_toto_builder.from_manifest
+
+    # Put everything together
+    if not args.dry_run:
+        in_toto_builder(serializer.serialize(args.path))
+
+
+def build_parser() -> argparse.ArgumentParser:
+    """Builds the command line parser for the bechmark runner."""
+    parser = argparse.ArgumentParser(
+        description="Benchmark full serialization of a model for model signing"
+    )
+
+    parser.add_argument("path", help="path to model", type=pathlib.Path)
```

Comment:
your `generate.py` had some args as `required`. It seems `path` may be one we want to make required here as well?

Reply:
It is required because it does not start with `-`/`--` (it's not an option, it's a positional argument)